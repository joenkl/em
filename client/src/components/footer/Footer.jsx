import React from 'react';
import _ from 'lodash';
import Grid from '@material-ui/core/Grid';
import { makeStyles, useTheme } from '@material-ui/styles';

import TEXT from '../../config/text';
const FOOTER_TEXT = _.get(TEXT, 'FOOTER');

const useStyles = makeStyles({
  footer: theme => ({
    display: 'block',
    color: _.get(theme, 'palette.primary.light'),
    padding: 16,
    marginTop: 'auto',
  }),
  about: { },
  terms: { },
  policy: { },
});

const Footer = (props) => {
  const theme = useTheme();
  const classes = useStyles(theme);
  return (
    <footer className={classes.footer}>
      <Grid
        container
        direction="row"
        justify="center"
        alignItems="flex-end"
      >
        <Grid items xs className={classes.about}>
          {FOOTER_TEXT.ABOUT}
        </Grid>
        <Grid items xs className={classes.terms}>
          {FOOTER_TEXT.TERMS}
        </Grid>
        <Grid items xs className={classes.policy}>
          {FOOTER_TEXT.PRIVACY_POLICY}
        </Grid>
      </Grid>
      <Grid item xs>
        {FOOTER_TEXT.COPYRIGHT}
      </Grid>
    </footer>
  )
}

export default Footer