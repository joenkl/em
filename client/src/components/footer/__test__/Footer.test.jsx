import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Footer from '../Footer';

test('Home page should render correctly', () => {
  const elem = ReactTestRenderer.create(
      <Footer />
  )

  const tree = elem.toJSON()
  expect(tree).toMatchSnapshot()
})