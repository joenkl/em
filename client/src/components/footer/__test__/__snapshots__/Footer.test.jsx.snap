// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correctly 1`] = `
<footer
  className="Footer"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root about MuiGrid-grid-xs-true"
      items={true}
    >
      About
    </div>
    <div
      className="MuiGrid-root terms MuiGrid-grid-xs-true"
      items={true}
    >
      Terms
    </div>
    <div
      className="MuiGrid-root policy MuiGrid-grid-xs-true"
      items={true}
    >
      Privacy Policy
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
  >
    Jatda Studio Â© 2019
  </div>
</footer>
`;
